tests/run_all.sh

```diff
@@ -1,6 +1,12 @@
 #!/bin/bash
 
+# terminate after the first line that returns non-zero exit code
+set -e
+
 export TEST_DIR="$PWD/tests"
+tests_to_run=(img_test img_test_png base_test)
 
-./main.sh -r . -c tests/img_test.in | diff tests/img_test.out -
-./main.sh -r . -c tests/base_test.in | diff tests/base_test.out -
+for test in "${tests_to_run[@]}"; do
+  echo "$test"
+  ./main.sh -r . -c "tests/${test}.in" | diff "tests/${test}.out" -
```

Comment:
can you just use the exact one I have on the other branch? IMO it's also useful to be able to run all the tests at once, not just terminate if anything fails.

Reply:
You mean this?

```bash
for test in "${tests_to_run[@]}"; do
	echo "$test"
	./main.sh -r . -c "tests/${test}.in" | diff "tests/${test}.out" - && echo "Pass"
done
```

Correct me if I'm wrong, but I think that will always make the build pass (even if tests are failing), since the exit code that is returned will always be `0`. You'd have to check the exit code at the end of each diff and then set a flag to return a non-zero exit code if a test fails.